contrib/golang/filters/http/source/golang_filter.cc

```diff
@@ -157,21 +159,67 @@ void Filter::onDestroy() {
 // access_log is executed before the log of the stream filter
 void Filter::log(const Formatter::HttpFormatterContext& log_context,
                  const StreamInfo::StreamInfo&) {
+  uint64_t req_header_num = 0;
+  uint64_t req_header_bytes = 0;
+  uint64_t req_trailer_num = 0;
+  uint64_t req_trailer_bytes = 0;
+  uint64_t resp_header_num = 0;
+  uint64_t resp_header_bytes = 0;
+  uint64_t resp_trailer_num = 0;
+  uint64_t resp_trailer_bytes = 0;
+
+  auto decoding_s = dynamic_cast<processState*>(&decoding_state_);
+  auto encoding_s = dynamic_cast<processState*>(&encoding_state_);
+  auto prev_decoding_filter_state = decoding_state_.filterState();
+  auto prev_encoding_filter_state = encoding_state_.filterState();
+
   // `log` may be called multiple times with different log type
   switch (log_context.accessLogType()) {
   case Envoy::AccessLog::AccessLogType::DownstreamStart:
   case Envoy::AccessLog::AccessLogType::DownstreamPeriodic:
   case Envoy::AccessLog::AccessLogType::DownstreamEnd:
     // log called by AccessLogDownstreamStart will happen before doHeaders
     if (initRequest()) {
-      request_headers_ = static_cast<Http::RequestOrResponseHeaderMap*>(
-          const_cast<Http::RequestHeaderMap*>(&log_context.requestHeaders()));
+      request_headers_ = const_cast<Http::RequestHeaderMap*>(&log_context.requestHeaders());
     }
 
     // This only run in the work thread, it's safe even without lock.
-    is_golang_processing_log_ = true;
-    dynamic_lib_->envoyGoFilterOnHttpLog(req_, int(log_context.accessLogType()));
-    is_golang_processing_log_ = false;
+    decoding_state_.setFilterState(FilterState::ProcessingLog);
```

Comment:
@doujiang24 
What about adding a pair of new methods:
* setProcessingLog
* isProcessingLog

So we can check if we are processing Go in the instance of ProcessorState.

Reply:
Do you mean still write the logging state into `decoding/encoding_state`?
I don't think it is a good idea, logging state does not belongs to `decoding/encoding`, it would be some kind of hacky and make the code a bit complicated.
A new flag could be much clear.